alns/accept/HillClimbing.py

```diff
@@ -1,11 +1,58 @@
+from collections import deque
+from typing import List, Optional
+
 from alns.accept.AcceptanceCriterion import AcceptanceCriterion
 
 
 class HillClimbing(AcceptanceCriterion):
     """
-    Hill climbing only accepts progressively better solutions, discarding those
-    that result in a worse objective value.
+    The Hill Climbing (HC) criterion only accepts progressively better solutions,
+    discarding those that result in a worse objective value.
+
+    There are three variants of the HC criterion:
+    1) Classic (HC): only accepts a candidate solution if it is better
+       than the current solution.
+    2) Late Acceptance (LAHC): only accepts a candidate solution if it is
+       better than the current solution from a number of iterations ago.
+    3) Improved LA (ILAHC): this variant combines HC and LAHC.
```

Comment:
Question: how does it combine the two? [we should probably explain that]

Reply:
Answer:
```
1) HillClimbing()
2) HillClimbing(False, some_number)
3) HillClimbing(n_last=some_number)
```
But we should probably change this. See https://github.com/N-Wouda/ALNS/pull/76#discussion_r889680149.